File: Python/204__Count_Primes.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n <= 1:
            return 0

        isPrime = [True] * n
        isPrime[0] = False

        for i in xrange(2, n):
            if i * i > n:
                break;
            if not isPrime[i]:
                continue;
            for j in xrange(i * i, n, i):
                isPrime[j] = False

        count = 0

        for i in xrange(2, n):
            if isPrime[i]:
                count += 1

        return count

File: Java/204__Count_Primes.java
class Solution {
    public int countPrimes(int n) {
        if (n == 0 || n == 1) {
            return 0;
        }
        int result = 0;
        boolean[] isPrime = new boolean[n];
        Arrays.fill(isPrime, true);
        isPrime[0] = false;
        isPrime[1] = false;
        for (int p = 2; p < Math.sqrt(n); p++) {
            if (isPrime[p]) {
                for (int j = p * p; j < n; j += p) {
                    isPrime[j] = false;
                }
            }
        }
        for (int i = 2;i < isPrime.length; i++)
            if (isPrime[i]) {
                result++;
            }
        return result;
    }
}
